src/Bridge/Laravel/Http/Middlewares/AuditMiddleware.php

```diff
@@ -0,0 +1,181 @@
+        if (($originalException instanceof Exception) === true) {
+            throw $originalException;
+        }
+
+        return $response;
+    }
+
+    /**
+     * Call the HttpLoggerInterface to log psr request and response
+     *
+     * @param \DateTime $datetime
+     * @param string $ip
+     * @param \Psr\Http\Message\RequestInterface|null $psrRequest
+     * @param \Psr\Http\Message\ResponseInterface|null $psrResponse
+     *
+     * @return void
+     */
+    private function callHttpLogger(
+        BaseDateTime $datetime,
+        string $ip,
+        ?RequestInterface $psrRequest,
+        ?ResponseInterface $psrResponse = null
+    ): void {
+        try {
+            if (($psrRequest instanceof RequestInterface) === true) {
+                $this->httpLogger->record(
+                    $ip,
+                    $datetime,
+                    $psrRequest,
+                    $psrResponse
+                );
+            }
+        } catch (Exception $exception) {
+            $this->logger->exception($exception);
+        }
+    }
+
+    /**
+     * Convert symfony request to PSR7 request
+     *
+     * @param \Illuminate\Http\Request $request
+     *
+     * @return \Psr\Http\Message\ServerRequestInterface|null
+     */
+    private function createPsr7Request(Request $request): ?ServerRequestInterface
+    {
+        $psrRequest = null;
+        try {
+            $psrRequest = $this->psr7Factory->createRequest($request);
+        } catch (Exception $exception) {
+            $this->logger->exception($exception);
+        }
+
+        return $psrRequest;
+    }
+
+    /**
+     * Convert symfony response to PSR7 response
+     *
+     * @param \Symfony\Component\HttpFoundation\Response $response
+     *
+     * @return \Psr\Http\Message\ResponseInterface|null
+     */
+    private function createPsr7Response(Response $response): ?ResponseInterface
+    {
+        $psrResponse = null;
+
+        try {
+            $psrResponse = $this->psr7Factory->createResponse($response);
+        } catch (Exception $exception) {
+            $this->logger->exception($exception);
+        }
+
+        return $psrResponse;
+    }
+
+    /**
+     * @param \EoneoPay\Utils\DateTime $dateTime
+     * @param \Illuminate\Http\Request $request
+     * @param \Symfony\Component\HttpFoundation\Response|\Psr\Http\Message\ResponseInterface|null $response
+     *
+     * @return void
+     */
+    private function processForLogging(
+        DateTime $dateTime,
+        Request $request,
+        $response = null
+    ): void {
+        $psrRequest = $this->createPsr7Request($request);
+
+        $psrResponse = null;
+        if (($response instanceof Response) === true) {
+            $psrResponse = $this->createPsr7Response($response);
+        }
+
+        // if response is already an instance of psr7
+        if (($response instanceof ResponseInterface) === true) {
+            $psrResponse = $response;
+        }
```

Comment:
What happens here if Response is something else entirely?

Reply:
the HttpLoggerInterface is able to accept a null psr response.

Reply:
Good - but what happens if the next middleware layer returns something it really shouldn't?

Reply:
If the response is anything that cannot be converted to PSR response, shouldn't it just be null?

Reply:
Going to have to be. At this point, we've already done all the work and committed things to the database, we can't just not do anything with it.
Write a test for this scenario, and then we can decide in #eoneopay channel how to handle it (which will probably be log in and don't die horribly)

Reply:
good thing. Will do.